### PR TITLE
More pyRenderTarget

### PR DESCRIPTION
**Pull request by Hoikas**
This does several things to the pyRenderTarget bindings:

- Splits pyDynamicCamMap and pyCubicRenderTarget into their own files. It's kind of crazy to have multiple Python bindings in one file.
- Fixes potential access violations in pyCubicRenderTarget
- Adds sequence manipulators to pyDynamicCamMap to be consistent with the rest of PyHSPlasma
- Adds getters and setters for all plRenderTarget members to pyRenderTarget

### CODE REVIEW

**Comment by Hoikas** [comment]
I'm finding some bugs in other areas of pyRenderTarget unrelated to this PR. I will update when they're fixed as well.

**Comment by Hoikas** [comment]
Done fixing tangential bugs! :smile:

File: Python/PRP/Surface/pyCubicRenderTarget.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    pyCubicRenderTarget* self = (pyCubicRenderTarget*)type->tp_alloc(type, 0);
+    if (self != NULL) {
+        self->fThis = new plCubicRenderTarget();
+        self->fPyOwned = true;
+    }
+    return (PyObject*)self;
+}
+
+static PyObject* pyCubicRenderTarget_getLeftFace(pyCubicRenderTarget* self, void*) {
+    return pyRenderTarget_FromRenderTarget(IConvertRT(self)->getFace(plCubicRenderTarget::Faces::kLeftFace));
```
Comment (zrax) [inline Python/PRP/Surface/pyCubicRenderTarget.cpp]: Non-standard C++:  `enum Faces` is not an `enum class`, so these should be `plCubicRenderTarget::kLeftFace`, etc.

File: Python/PRP/Surface/pyCubicRenderTarget.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+static PyGetSetDef pyCubicRenderTarget_GetSet[] = {
+    { _pycs("leftFace"), (getter)pyCubicRenderTarget_getLeftFace, NULL, NULL, NULL },
+    { _pycs("rightFace"), (getter)pyCubicRenderTarget_getRightFace, NULL, NULL, NULL },
+    { _pycs("frontFace"), (getter)pyCubicRenderTarget_getFrontFace, NULL, NULL, NULL },
+    { _pycs("backFace"), (getter)pyCubicRenderTarget_getBackFace, NULL, NULL, NULL },
+    { _pycs("topFace"), (getter)pyCubicRenderTarget_getTopFace, NULL, NULL, NULL },
+    { _pycs("bottomFace"), (getter)pyCubicRenderTarget_getBottomFace, NULL, NULL, NULL },
+    { NULL, NULL, NULL, NULL, NULL }
```
Comment (zrax) [inline Python/PRP/Surface/pyCubicRenderTarget.cpp]: No generic `getFaces` or faces iterable?

File: Python/PRP/Surface/pyDynamicCamMap.cpp [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+    }
+    self->fThis->delTargetNode((size_t)idx);
+    Py_INCREF(Py_None);
+    return Py_None;
+}
+
+static PyObject* pyDynamicCamMap_addVisRegion(pyDynamicCamMap* self, PyObject* args) {
+    PyObject* key;
+    if (!(PyArg_ParseTuple(args, "O", &key) && pyKey_Check(key))) {
+        PyErr_SetString(PyExc_TypeError, "addVisRegions expects a plKey");
```
Comment (zrax) [inline Python/PRP/Surface/pyDynamicCamMap.cpp]: addVisRegion*

File: Python/PRP/Surface/pyDynamicCamMap.cpp [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+static PyObject* pyDynamicCamMap_clearVisRegions(pyDynamicCamMap* self) {
+    self->fThis->clearVisRegions();
+    Py_INCREF(Py_None);
+    return Py_None;
+}
+
+static PyObject* pyDynamicCamMap_delVisRegion(pyDynamicCamMap* self, PyObject* args) {
+    Py_ssize_t idx;
+    if (!PyArg_ParseTuple(args, "n", &idx)) {
+        PyErr_SetString(PyExc_TypeError, "delVisRegions expects an int");
```
Comment (zrax) [inline Python/PRP/Surface/pyDynamicCamMap.cpp]: delVisRegion*

File: Python/PRP/Surface/pyCubicRenderTarget.cpp (hunk 2 above)
Comment (Hoikas) [inline Python/PRP/Surface/pyCubicRenderTarget.cpp]: This was just moved from another file, but I can add that if you want.

File: Python/PRP/Surface/pyDynamicCamMap.cpp [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+    const std::vector<plKey>& keys = self->fThis->getVisRegions();
+    PyObject* regionList = PyList_New(keys.size());
+    for (size_t i=0; i<keys.size(); i++)
+        PyList_SET_ITEM(regionList, i, pyKey_FromKey(keys[i]));
+    return regionList;
+}
+
+static PyObject* pyDynamicCamMap_getTargetNodes(pyDynamicCamMap* self, void*) {
+    const std::vector<plKey>& keys = self->fThis->getTargetNodes();
+    PyObject* regionList = PyList_New(keys.size());
```
Comment (zrax) [inline Python/PRP/Surface/pyDynamicCamMap.cpp]: Variable name copypasta

File: Python/PRP/Surface/pyDynamicCamMap.cpp [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+        return -1;
+    }
+}
+
+static PyMethodDef pyDynamicCamMap_Methods[] = {
+    { "addMatLayer", (PyCFunction)pyDynamicCamMap_addMatLayer, METH_VARARGS,
+      "Params: key\n"
+      "Adds a layer key" },
+    { "clearMatLayers", (PyCFunction)pyDynamicCamMap_clearMatLayers, METH_NOARGS,
+      "Clears the list of layerkeys" },
```
Comment (zrax) [inline Python/PRP/Surface/pyDynamicCamMap.cpp]: space?

File: Python/PRP/Surface/pyCubicRenderTarget.cpp (hunk 2 above)
Comment (zrax) [inline Python/PRP/Surface/pyCubicRenderTarget.cpp]: Ah; I guess only if you want to then :)

**Pull request merged**